wordpress-mobile/WordPress-iOS
# Posts: Moves blog post specific logic from BasePost into AbstractPost.

Closes #5523 
This PR just moves logic specific to AbstractPost to that class and out of BasePost.  I tried to organize the moved methods into related groups but not shuffle existing methods too much.  There are other opportunities for clean up but that's for another time. 

To test: 
Test the posts feature. Make sure you can sync, edit, etc without errors. 
Test the reader. Make sure you can sync and view post details without errors. 

Needs review: @diegoreymendez
